processing/src/main/java/io/druid/query/DefaultQueryMetrics.java

```diff
@@ -66,15 +65,6 @@ public void queryType(QueryType query)
   }
 
   @Override
-  public void interval(QueryType query)
```

Comment:
I understand what you're concerned with. For backward compatibility, I'll keep ```DefaultQueryMetrics``` as it is add a new ```MultiSourceDefaultQueryMetrics``` for join. Does it make sense?

With an assumption of QueryMetrics is generally used for monitoring and tuning query performance, I think ```dataSourcesAndIntervals()``` will be more useful than emitting dataSources and intervals separately for join. In this patch, intervals are not associated with a query anymore. Instead, they are associated with each data source. This is for allowing join of data sources of different intervals. The performance of a join query highly relies on the size of its inputs, and the ```intervals``` relate to how large data will be read from each data source. This is why I stick to ```dataSourcesAndIntervals()``` which emits data sources and corresponding intervals together.